test/extended/util/client.go

```diff
@@ -75,15 +75,15 @@ type CLI struct {
 	execPath               string
 	command                string
 	configPath             string
-	currentWs              WorkSpace
+	currentWorkSpace       *WorkSpace
```

Comment:
From my unsderstanding, as the `WorkSpace` struct size is small no need to use pointer here, use a value receiver is efficient and clear. It also suggested by golang offical FAQ.
https://go.dev/doc/faq#methods_on_values_or_pointers
```
For types such as basic types, slices, and small structs, a value receiver is very cheap so unless the semantics of the method requires a pointer, a value receiver is efficient and clear.
```

Reply:
HI @Phaow, I decided to use pointer to abstract out WorkSpace object into its own package. If we don't use pointers, an update on the workspace side will not be reflect on the `workSpacesToDelete`. To unify the operation, I set all workspace variables to pointers to ensure consistency.

Reply:
Hmm, for current logic maybe no update on the workspaces side, maybe it could use for we expand the worksapce struct methods. It's ok for me if you want to expand the struct usages but maybe a bit strange that we use the method `ws.GenerateNamespace`. Generally usage like `ws.Create` , `ws.Delete`. AFAIK, the ns is just a kind of reosurce under the ws like pod under ns in kubernetes.

Reply:
Hi @Phaow, I used pointer because `TeardownWorkSpace` also deletes namespaces under each workspace object. If `WorkSpace` is not a pointer, when the user add a new namespace using the `CreateNamespace` function, it will not be reflected in the `workSpacesToDelete` array, and the namespace will not be deleted, creating ns leftover.

Reply:
Hmm, delete the workspace will also deleted all its namespaces and current logic `TeardownWorkSpace` won't delete ns. If you want to keep it, I'm ok.

Reply:
@zxiao-redhat @Phaow thanks for discussions. From my perspective, using value or pointer is technically either fine.